### PR TITLE
Create a file tile component

### PR DESCRIPTION
**Pull request by LunarX**
For now the FileTile component relies on a temporary FileUiItem while we wait for the KMP model

### CODE REVIEW

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    isChecked: () -> Boolean = { false },
+    onClick: () -> Unit,
+    onRemove: (() -> Unit)? = null,
+) {
+    FileTileContent(
+        content = {
+            var displayPreview by rememberSaveable { mutableStateOf(file.hasPreview) }
+
+            if (displayPreview) {
+                FileThumbnail(file.uri.toUri()) { displayPreview = false }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt:77]: Add the callback name instead

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+interface FileUiItem {
+    val uid: String
+    val fileName: String
+    val fileSizeInBytes: Long
+    val mimeType: String?
+    val uri: String
+}
+
+@Composable
+fun FileTile(
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt:64]: Rename to `FileItem`

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    Card(
+        onClick = onClick,
+        modifier = Modifier.aspectRatio(164 / 152f),
+        colors = CardDefaults.cardColors(containerColor = SwissTransferTheme.materialColors.background),
+        shape = Shapes.small,
+        border = BorderStroke(width = 1.dp, SwissTransferTheme.materialColors.outlineVariant)
+    ) {
+        Box(
+            modifier = Modifier
+                .weight(1f)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt:112]: Check if `weight` is really needed ?

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -42,9 +50,18 @@ swisstransfer-core = { module = "com.github.Infomaniak.multiplatform-SwissTransf
 androidx-junit = { group = "androidx.test.ext", name = "junit", version.ref = "junitVersion" }
 junit = { group = "junit", name = "junit", version.ref = "junit" }
 
+# FileType module
+androidx-ui-android = { group = "androidx.compose.ui", name = "ui-android", version.ref = "uiAndroid" }
+androidx-foundation-android = { group = "androidx.compose.foundation", name = "foundation-android", version.ref = "foundationAndroid" }
+androidx-ui-tooling-preview-android = { group = "androidx.compose.ui", name = "ui-tooling-preview-android", version.ref = "uiToolingPreviewAndroid" }
+androidx-ui-tooling = { group = "androidx.compose.ui", name = "ui-tooling", version.ref = "uiTooling" }
+
```
Comment (sirambd) [inline gradle/libs.versions.toml:58]: You can extract it to own file and declare in `settings.gradle.kts` like :
```kt
dependencyResolutionManagement {
    repositories {
        google()
        mavenCentral()
    }
    versionCatalogs {
        create("libs") {
            from(files("../gradle/libs.versions.toml"))
        }
    }
}
```

**Review by sirambd — CHANGES_REQUESTED** [review]

File: FileTypes/src/main/java/com/infomaniak/library/filetypes/FileType.kt [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+import android.util.Log
+import android.webkit.MimeTypeMap
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import com.infomaniak.library.filetypes.icons.*
+import org.jetbrains.annotations.TestOnly
+
+private val MEDIA_COLOR_LIGHT = Color(0xFF00BCD4)
+private val MEDIA_COLOR_DARK = Color(0xFF86DEEA)
```
[the remark this answers is not in the document]
Comment (LunarX) [reply FileTypes/src/main/java/com/infomaniak/library/filetypes/FileType.kt:28]: Actually this seems impossible as the IDE warns us with an error when trying to access the constant inside the enum that says:

```
Companion object of enum class 'FileType' is uninitialized here
Variable 'MEDIA_COLOR_LIGHT' must be initialized
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt (hunk 3 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/components/FileTile.kt:112]: Yeah no, it's actually needed we didn't look carefully but this Box is one of the two childs of the Card's ColumnScope. Yeah it's needed

**Review by sirambd — APPROVED** [review]

**Pull request merged**